src/common/version.ts

```diff
@@ -34,6 +34,18 @@ function versionGte(a: VersionInfo, b: VersionInfo): boolean {
   return false;
 }
 
+/**
+ * The minimum version of the Ruff executable that supports the native server.
+ */
+export const MINIMUM_SUPPORTED_EXECUTABLE_VERSION: VersionInfo = { major: 0, minor: 1, patch: 0 };
```

Comment:
Which version is this? I don't think it's the one as described in the doc.

Reply:
it's first minor released version. I'm open to change it to higher.

Reply:
Oh, it _is_ for the native server, but we already have a check for that using `MINIMUM_NATIVE_SERVER_VERSION` variable:

https://github.com/astral-sh/ruff-vscode/blob/8f0e0c308d810dbb9d4ff68469db8c9b2109d279/src/common/server.ts#L181-L188

So, this isn't required here.

Reply:
(`MINIMUM_SUPPORTED_EXECUTABLE_VERSION` added in https://github.com/astral-sh/ruff-vscode/pull/585/commits/81f9ea2d9461e4e94375b0c31adfe3555d96672e to cover better error message for https://github.com/astral-sh/ruff-vscode/issues/323)

Reply:
Oh right, that is actually coming from `ruff-lsp` and isn't related to this part of the code:

https://github.com/astral-sh/ruff-lsp/blob/2bd56d956d3f583429b78fa050f124515732cb49/ruff_lsp/server.py#L1761-L1763

So, we should remove this. This code is only run for the native server so I don't think it makes sense to tackle that issue.

Reply:
Sorry, I'm still getting confused, I think still it'd make sense to declare minimum supported version in here:
1. because https://github.com/astral-sh/ruff-lsp/blob/2bd56d956d3f583429b78fa050f124515732cb49/ruff_lsp/server.py#L1761-L1763 will fail after chosen server kind and there will no fallback from ruff-vscode side.
2. it may need re-define it in future to drop deprecated settings.

Reply:
I think in your testing, you can consider test with and without this change via exposing ruff < 0.1.0 in your PATH.
I think without this change it most unable fallback to bundled version.

Reply:
Sorry for the confusion. What I'm saying is that this code path is only taken for the native server and not when the user has selected the `ruff-lsp`. The `ruff-lsp` server chooses the binary in the server code itself while for the native server, the VS Code extension is responsible for choosing the binary.

Reply:
Oh, this error will be logged in case the `ruff.nativeServer` value is "auto" because we might try to find the Ruff binary to decide whether to use the native server or the Python based server.

Reply:
But, I don't think it's right for that message to be an error because in the "auto" behavior, the extension is trying to find the best executable based on the user setting and choosing between the native server and `ruff-lsp`. I'll update the message and use info level instead.

Reply:
Also, this doesn't really resolve the linked issue because for `ruff.nativeServer: "off"` it still persists:

<img width="1727" alt="Screenshot 2024-08-26 at 10 54 04" src="https://github.com/user-attachments/assets/3921ed2f-2270-43e7-a6af-d6adbb19485b">

Reply:
> Sorry for the confusion. What I'm saying is that this code path is only taken for the native server and not when the user has selected the `ruff-lsp`.

Now I got it. I think it's because `findRuffBinaryPath` only used by native server resolution functions. Since ruff-lsp has its own binary locator, I think this change should sync in there too.

> The `ruff-lsp` server chooses the binary in the server code itself while for the native server, the VS Code extension is responsible for choosing the binary.
I think it makes sense to choosing binary for ruff-lsp in vscode extension side and pass the found binary to ruff-lsp via cli? what do you think?

Reply:
> Also, this doesn't really resolve the linked issue because for `ruff.nativeServer: "off"` it still persists:

For now, I'm going to add `findRuffBinaryPath` in `createLegacyServer` to prepare `ruff-lsp` for syncing executable locator code.